base/bitarray.jl

```diff
@@ -506,20 +506,23 @@ function BitArray{N}(A::AbstractArray{T,N}) where N where T
     Bc = B.chunks
     l = length(B)
     l == 0 && return B
-    ind = 1
+    Aind = eachindex(A)
```

Comment:
Local bench shows that this change makes this constructor almost 6x slower if `A` is linear-indexable. 
Should we move the copy part (L509~L528) to, e.g. `copyto!(dest::BitArray, src::AbstractArray)` and optimization for `src` with different `IndexStyle`?

Reply:
What benchmark did you run?

Reply:
```julia
julia> a = view(rand(Bool, 100, 100), :, :);

julia> @btime BitArray_master($a);
  1.340 μs (2 allocations: 1.38 KiB)

julia> @btime BitArray_pr($a);
  9.200 μs (2 allocations: 1.38 KiB)
```
`@code_llvm` shows that this PR failed to vectorize due to union split.